File: task1/task1c/src/main/java/com/task1c/SeaVessel.java
package com.task1c;

import java.util.ArrayList;


public class SeaVessel extends Transport{
    private ArrayList<String> routeList = new ArrayList<String>();
    private String route;

    public SeaVessel(){

    }

    public SeaVessel(int id, String type, double speed, double cost){
        super(id, type, speed, cost);
    }

    @Override
    public void display_vehicle_details(){
        super.display_vehicle_details();
        System.out.println("Route List:");
        for (int i = 0; i < routeList.size(); i++){
            System.out.println(routeList.get(i));
        }
    }

    // Setters
    public void setRoute(String route){
        this.route = route;
    }

    public void addToList(String route){
        routeList.add(route);
    }

    public void updateRouteinList(String route, int pos){
        routeList.set(pos, route);
    }

    // Getters
    public String getRoute(int pos){
        String routeAtPos = routeList.get(pos);
        return routeAtPos;
    }
}

File: task1/task1b/SeaVessel.cpp
#include "SeaVessel.h"
#include <iostream>
#include <string.h>

SeaVessel::SeaVessel(){}

SeaVessel::SeaVessel(int id, const string& type, double speed, double cost): Transport(id, type, speed, cost) {}

void SeaVessel::display_transport_details() {
    Transport::display_transport_details();
    cout << "Route List: " << endl;
    for(int i = 0; i < routeList.size(); i++){
        cout << routeList.at(i) << endl;
    }
}

void SeaVessel::updateRouteinList(const string& route, int pos){
    routeList.at(pos) = route;
}


// Setters
void SeaVessel::setRoute(const string& route){
    this->route = route;
}

void SeaVessel::addToList(const string& route){
    routeList.push_back(route);
}


// Getters
string SeaVessel::getRoute(int pos) const{
    string routeAtPos = routeList.at(pos);
    return routeAtPos;
}